pkg/assets/builder_test.go

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package assets
+
+import (
+	"k8s.io/kops/pkg/apis/kops"
+	"testing"
+)
+
+func TestRemap_File(t *testing.T) {
+	grid := []struct {
+		testFile   string
+		expected   string
+		asset      *FileAsset
+		kopsAssets *kops.Assets
+	}{
+		{
+			// FIXME - need https://s3.amazonaws.com/k8s-for-greeks-kops/kubernetes-release/release/v1.7.2/bin/linux/amd64/kubelet
```

Comment:
We do, but this only applies to nodeup, right?  Because AIUI once we are in nodeup we can (more) easily download from a secured bucket.  But nodeup is tricky, and we either want to remap or use a signed URL.

We can also remap on the upload, i.e. we can recognize that a write to `https://s3.amazonaws.com/k8s-for-greeks-kops/` becomes a write to `s3://k8s-for-greeks-kops/`, using a public ACL.

To my mind, a signed URL is going to be best, because I don't think people that want to run mirrors are going to love the idea of public buckets.

Not sure if this FIXME is a blocker for merge?  I'm guessing not, because it doesn't cause any regression in existing behaviour (i.e. this doesn't work yet anyway)?